appserver/ejb/ejb-container/src/main/java/org/glassfish/ejb/deployment/annotation/handlers/AbstractEjbHandler.java

```diff
@@ -233,9 +233,9 @@ public HandlerProcessingResult processAnnotation(AnnotationInfo ainfo)
                                    ejbClass.getName() }));
                 return getDefaultFailedResult();
             }
-
-
-        } else {
+        }
+        // Create a new descriptor only if there are no entries in the ejb-jar with same class name. EJB 3.0 spec
+        else if(currentBundle.getEjbByClassName(ejbClass.getName()).length == 0) {
```

Comment:
In that particular test case you showed, it has an ejb-jar.xml which defines this EJB. So it should fall into the if block before this else block as it should satisfy this condition?
if (ejbDesc != null && !(ejbDesc instanceof DummyEjbDescriptor) ) {
Trying to understand why changing code in the else block would affect the behavior of that test case.